### ISSUE TITLE
ssl issues on k1c

### ISSUE THREAD

**Issue by Moonman0922** [ballaswag/guppyflo#9]
I believe I am experiencing ssl certificate issues when installing on my creality k1c. The console output is as follows:
```shell
sh installer.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (60) SSL certificate problem: certificate is not yet valid
More details here: https://curl.haxx.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
unzip: can't open /tmp/guppyflo.zip[.zip]
Configuring GuppyFLO services as HTTP Reverse Proxy
Restarting GuppyFLO service. Please wait...

Successfully installed GuppyFLO
``` 
I don't see the tailscale url and I cannot access the webui.

**Comment by Moonman0922** [comment]
Ok, so this took me quite a while to figure out how to solve. 

The SSL certificate issue was caused by the system clock resetting back to the factory time and date constantly. This was solved by following [this Reddit thread](https://www.reddit.com/r/crealityk1/comments/16f6rgu/rooted_k1_always_resetting_time_to_mar_1/) and installing chrony. 

After having manually set the time and date as close as possible using the code below.

```shell
date +%Y%m%d -s "20081128"
date +%T -s "10:13:13"
``` 
This allowed me to get close enough to synchronized to install the K1 [Creality-Helper-Script](https://github.com/Guilouz/Creality-Helper-Script-Wiki). 

I installed moonraker, nginx, guppyflo, and entware. This worked wonderfully until I rebooted the printer and Tailscale was no longer working. When I checked the log file, it was caused by another certificate problem. At this point I installed chrony as stated above, and everything is now working. 

I still don't know why the time and date issues are recurring, but this solution worked for me for now. I just wanted to make sure the next person to deal with this has more to go off of!

I have copied the information in the Reddit thread below just in case it's not accessible later!
```shell
u/Garlic_Emergency

#here is how to install

#Installing Entware and ChronyFirst, you need to install Entware using Guilouz's script. 
#Follow the instructions on the GitHub page linked in the script documentation. 
#Once Entware is installed, you can install Chrony with the following command:

>$ opkg install chrony

#**Configure Chrony**
#Create the Chrony configuration file with the server settings you've listed. 
#You can do this by entering these commands: 
>$ mkdir -p /usr/data/opt/etc/chrony
>$ touch /usr/data/opt/etc/chrony/chrony.conf

#Now, you need to edit the chrony.conf file. 
#You can use a text editor available in your Linux distribution, such as nano or vi. 
#For example, to edit with nano:nano /usr/data/opt/etc/chrony/chrony.conf
#Then, paste your configuration into the editor:
#Use public NTP servers from the [pool.ntp.org](http://pool.ntp.org/) project. 

>$ pool [pool.ntp.org](http://pool.ntp.org/) iburst
>$ server [a.st1.ntp.br](http://a.st1.ntp.br/) iburst
>$ server [b.st1.ntp.br](http://b.st1.ntp.br/) iburst
>$ server [c.st1.ntp.br](http://c.st1.ntp.br/) iburst
>$ server [d.st1.ntp.br](http://d.st1.ntp.br/) iburst
>$ server [gps.ntp.br](http://gps.ntp.br/) iburst

#[The Reddit thread was designed for someone in Brazil. 
#If you use the following servers will auto-select the best for your region! -QM]

>$ server 0.pool.ntp.org
>$ server 1.pool.ntp.org
>$ server 2.pool.ntp.org
>$ server 3.pool.ntp.org

#Record the rate at which the system clock gains/losses time.

>$ driftfile /var/lib/chrony/drift
>$ ntsdumpdir /var/lib/chrony

#Allow the system clock to be stepped in the first three updates if its offset is larger than 1 second.

>$ makestep 1.0 3

#Enable kernel synchronization of the real-time clock (RTC).

>$ rtcsync

#Save and exit the editor

#**Copy Chrony Initialization Script**

#Copy the Chrony initialization script from /usr/data/opt/etc/init.d to /etc/init.d to ensure it starts at boot:

>$ cp /usr/data/opt/etc/init.d/S06chronyd /etc/init.d/S06chronyd

#**Start Chrony Service**

#Now, start the Chrony service with:

>$ bash/etc/init.d/S06chronyd start

#**Verify Chrony is Working**

#To check if Chrony is synchronizing the time correctly, use:

>$ chronyc tracking

#This command will show you the status of the time synchronization.

#**Additional Tips**

#Make sure you have network connectivity as Chrony needs to reach the NTP servers.
#Ensure your device’s date and time are not drastically incorrect before starting, as it might affect Chrony’s ability to #synchronize.
#If you encounter permissions issues, you might need to run some commands with sudo if you're not logged in as root.

#By following these steps, you should have a working Chrony setup on your device to keep the time synchronized. If you run #into any specific errors during this process, feel free to ask for further help!
```

**Comment by Moonman0922** [comment]
Alright, so one manual power toggle later, and back to square one. I might have to add a script to install the fix above on every boot, as it doesn't look like it's sticking. Reopening issue until I can find a fix.


EDIT: Spoke too soon. Closing again, but with the added info that apparently it can take an absurdly long time to correct itself?

I powered on the printer, and sent a file over the non-Tailnet LAN connection. That file was timestamped with the incorrect 03/2020 date, and I was unable to access the webui for any tools. 

I accidentally clicked my bookmark link for my Fluidd webui instead of my Tailscale admin console, and it was working! This is after about 10-15 minutes since powering on from fully unplugged. Just an update if anyone else has intermittent issues!
